presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -370,6 +371,23 @@ private PlanBuilder explicitCoercionSymbols(PlanBuilder subPlan, Iterable<Symbol
                 analysis.getParameters());
     }
 
+    private List<Expression> constructProjectExpressions(QuerySpecification node)
```

Comment:
Good point.  The current implementation is consistent with the implementation of distinct.  I think it might be better to use a separate change to move both out of the scope to be consistent.